### PR TITLE
fix: fix addTokens call with an already added tokens to state

### PR DESCRIPTION
**Pull request by sahar-fehri**
## Explanation

When we first call the function `addTokens` with an array of tokens to import, and then call it again with the exact same token, we notice that `state.allTokens` is updated with duplicate entries of the same token.

`state.allTokens` are stored in checksum format.
When initially state already has USDC token for a user address; and a call to `addTokens` is made with a lowercase USDC address. The newTokensMap in this line https://github.com/MetaMask/core/blob/9f480c8ac7c88bc434c6bcd80dc7f0c58d0c1698/packages/assets-controllers/src/TokensController.ts#L514 is computed without checking if the value already exists in state.

The fix converts the tokensToImport received in the call to the checksum format.

## Checklist

- [ ] I've updated the test suite for new or updated code as appropriate
- [ ] I've updated documentation (JSDoc, Markdown, etc.) for new or updated code as appropriate
- [ ] I've communicated my changes to consumers by [updating changelogs for packages I've changed](https://github.com/MetaMask/core/tree/main/docs/contributing.md#updating-changelogs), highlighting breaking changes as necessary
- [ ] I've prepared draft pull requests for clients and consumer packages to resolve any breaking changes

---

> [!NOTE]
> Normalize token addresses to checksum in `TokensController.addTokens` to dedupe and avoid duplicate entries when the same token is added multiple times.
> 
> - **Assets Controllers**:
>   - **TokensController**: Normalize `tokensToImport` addresses to checksum and use checksummed list for deduping/updates in `addTokens`, preventing duplicate entries in `state.allTokens`.
> - **Tests**:
>   - Add unit test ensuring duplicate tokens are not re-added when the same token is imported twice (case-insensitive).
> - **Changelog**:
>   - Add Unreleased fix entry documenting duplicate token prevention in `state.allTokens`.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 6df8521e7b169c6e4a09dbab011370d0ace9af95. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: packages/assets-controllers/src/TokensController.test.ts [hunk 1]
```diff
@@ -1878,6 +1878,71 @@ describe('TokensController', () => {
         },
       );
     });
+
+    it('should not add duplicate tokens to state', async () => {
```
Comment (sahar-fehri) [inline packages/assets-controllers/src/TokensController.test.ts:1882]: This test fails on main and `state.allTokens` ends up with duplicate entries of the same token
